File: frontend/src/objects/rider/apiTypes/RiderMeResponse.ts
import { UserRoles } from '@/objects/shared/ids'
import type { UserRole } from '@/objects/shared/ids'
import type { Order } from '@/objects/order/Order'
import type { RiderAccountPublic } from '../RiderAccountPublic'
import type { RiderDeliveryStatus } from '../RiderDeliveryStatus'

export interface RiderMeResponse {
  username: string
  role: typeof UserRoles.rider & UserRole
  riderAccount: RiderAccountPublic
  availableOrders: Order[]
  deliveryStatuses: RiderDeliveryStatus[]
}

File: backend/src/rider/objects/apiTypes/RiderMeResponse.scala
package delivery.rider.objects.apiTypes

import delivery.order.objects.Order
import delivery.review.objects.{ReviewSummary, RiderReview}
import delivery.rider.objects.{RiderAccountPublic, RiderDeliveryStatus}
import delivery.shared.objects.UserRole

final case class RiderMeResponse(
    username: String,
    role: UserRole,
    riderAccount: RiderAccountPublic,
    availableOrders: List[Order],
    deliveryStatuses: List[RiderDeliveryStatus] = Nil,
    reviewSummary: ReviewSummary = ReviewSummary(5.0, 0),
    reviews: List[RiderReview] = Nil
)
